datamade/committee-oversight
# CHP Scores

## Overview

Closes #102. This PR follows the method described in #102 to implement a new `load_committeeratings` with the CHP score methodology, then use a variety of methods on the `CommitteeOrganization` and `CommitteeRatings` models to populate the data on the site. The testing instructions below include directions to download a data dump with (limited) categorization data entered by Lugar. 

There are a bunch of calculations going on here that can get confusing with the partial data we have—I didn't realize how few hearings have categorizations on the staging site, which I'm going to bring up with Lugar next week. For now, I wanted to pause here to get feedback on this approach.

## Testing Instructions

 * We need hearings data with categories to test the ratings system, so we can't keep using the sample data previously in `hearings.dump`. Go to the S3 bucket with backups from the hearings site, download the most recent (where I've deleted the committeeratings), and save it in this project's root as a new `hearings.dump`: https://s3.console.aws.amazon.com/s3/object/datamade-postgresql-backups/hearings/201911131730.dump?region=us-east-1&tab=overview
 Since this file is so large, I'm not keeping it under version control.
* Run `docker-compose down --volumes`
* Follow setup instructions in the README
* Go to http://localhost:8000/ and view the CHP scores you've just calculated. Verify that there are hearing counts attached to surveys.
* Visit the page for the [House Committee on Science, Space, and Technology](http://localhost:8000/committee-61caba6d-71bc-41c7-882e-32ec97768f63) and make sure the detail page is populating as you'd expect
* Visit the two comparison pages linked from the landing page
